### PR TITLE
Feat/update party importer

### PR DESCRIPTION
**Pull request by Bekabyx**
This is the underpinning work for phase 2 of party pages on Who ([Trello](https://trello.com/c/gziXi9a7/3366-wcivf-party-pages-part-ii)) 

There's a few things going on here which are mostly detailed in ref #2050, with an additional chunk to save the EC's json return against a party.

Test 1: Party alternative name
- Run the migrations
- Run `parties_import_from_ec`
- Hit the YNR API locally `http://localhost:8000/api/next/parties/{{party_id}}` (I recommend PP53 because they have a registered alt. name)
- Check the alternative name field - it should have a value if you've chosen a party who has an alternative name, or null if you haven't

Test 2: JSON field
- You've already run the migration and the mgmt command, so simply check any party in your database. the `ec_data` field should be populated.

Test 3: Emblems
- Manually add an emblem for a party - I did so with SQL but you can use any means you like. 
```sql
-- example
insert into parties_partyemblem(created,modified,"default",ec_emblem_id,image,description,party_id)
values (CURRENT_TIMESTAMP,CURRENT_TIMESTAMP,false,1337,'emblems/PP52/48_Emblem_48.jpg','test emblem',958);
```
- Run `parties_import_from_ec`
- The emblem you manually added should now have active set to false

Test 4: Descriptions
- Manually add a description for a party - I did so with SQL but you can use any means you like. 
```sql
-- example
insert into parties_partydescription(created,modified,description,date_description_approved,party_id)
values (CURRENT_TIMESTAMP,CURRENT_TIMESTAMP,'test description',CURRENT_TIMESTAMP,958);
```
- Run `parties_import_from_ec`
- The description you manually added should now have active set to false

```[tasklist]
### PR Checklist

- [x] References a specific issue or if not, describes the bug or feature in detail
- [x] Note what card in Trello this work relates to
- [x] Instructions for how reviewers can test the code locally
- [x] Tests have been added and/or updated
```

### CODE REVIEW

**Comment by VirginiaDooley** [comment]
Posting test results for transparency (feel free to remove if too bulky), but all good! 👏 

Test 1: 
Screenshot 2023-06-02 at 11 24 31 AM


Test 2: 
```
In [6]: Party.objects.last().ec_data
SELECT "parties_party"."id",
       "parties_party"."created",
       "parties_party"."modified",
       "parties_party"."ec_id",
       "parties_party"."name",
       "parties_party"."alternative_name",
       "parties_party"."register",
       "parties_party"."status",
       "parties_party"."date_registered",
       "parties_party"."date_deregistered",
       "parties_party"."legacy_slug",
       "parties_party"."current_candidates",
       "parties_party"."total_candidates",
       "parties_party"."ec_data"
  FROM "parties_party"
 ORDER BY "parties_party"."name" DESC
 LIMIT 1

Execution time: 0.004854s [Database: default]
Out[6]: 
{'Id': 344,
 'Town': 'London',
 'ECRef': 'PP344',
 'Line1': '50 Iverson Road',
 'Line2': 'West Hampstead',
 'Line3': None,
 'Line4': None,
 'County': None,
 'Country': 'United Kingdom',
 'Officers': [{'Name': 'Ms Eve Rea Juan',
   'Role': 'Leader',
   'RegulatedEntityId': 344,
   'RegulatedEntityRoleId': 1},
  {'Name': 'Ms Lumiere de Monde',
   'Role': 'Nominating Officer',
   'RegulatedEntityId': 344,
   'RegulatedEntityRoleId': 2},
  {'Name': 'Ms Lumiere de Monde',
   'Role': 'Treasurer',
   'RegulatedEntityId': 344,
   'RegulatedEntityRoleId': 3}],
 'Postcode': 'NW6 2HE',
 'LapsedDate': None,
 'RegisterId': 1,
 'IsLowerTier': None,
 'ApprovedDate': '/Date(1061852400000)/',
 'PartyEmblems': [],
 'ReferendumId': None,
 'RegisterName': 'Great Britain',
 'ReferendumName': None,
 'AccountingUnits': [],
 'CampaigningName': None,
 'EntityStatusName': None,
 'FinancialYearEnd': '31/12',
 'IsGibraltarParty': False,
 'PartyDescriptions': [],
 'ReferendumOutcome': None,
 'RegulatedEntityId': 344,
 'ThirdPermStatusId': None,
 'DesignationStatusId': None,
 'RegulatedEntityName': 'www.xat.org [De-registered 01/11/07]',
 'ThirdPermStatusName': None,
 'AssetLiabilityStatus': 0,
 'RegistrationStatusId': 2,
 'DesignationStatusName': None,
 'RegulatedEntityTypeId': 1,
 'ES_AnalysedQueryString': 'PP344 www.xat.org [De-registered 01/11/07] PP  Ms Eve Rea Juan Ms Lumiere de Monde Ms Lumiere de Monde',
 'RegisterNameMinorParty': None,
 'RegistrationStatusName': 'Deregistered',
 'RegulatedEntityStatusId': 16,
 'RegulatedEntityTypeName': 'Political Party',
 'ThirdPartyParticipators': None,
 'AssetLiabilityStatusName': 'N/A',
 'CompanyRegistrationNumber': None,
 'ES_NotAnalyzedQueryString': '"pp344" "www.xat.org_[de-registered_01/11/07]" "pp"  "ms_eve_rea_juan" "ms_lumiere_de_monde" "ms_lumiere_de_monde"',
 'FieldingCandidatesInWales': True,
 'LastIsLowerTierChangeDate': None,
 'RegulatedEntityStatusName': 'Statutorily Deregistered',
 'FieldingCandidatesInEurope': False,
 'FieldingCandidatesInEngland': True,
 'RegulatedEntityAbbreviation': 'PP',
 'FieldingCandidatesInScotland': True,
 'FieldingCandidatesMinorParty': False,
 'RegulatedEntityAlternateName': None,
 'RegulatedEntityTypeShortcode': 'PP',
 'TargetedSpendingThirdParties': [],
 'DisplayNIPartyReturnExemptions': False,
 'DateLastPartyNameChangeApproved': '/Date(1193875200000)/',
 'ExemptFromQuarterlyDonationReturns': True,
 'ExemptFromQuarterlyTransactionReturns': False,
 'ExemptFromParliamentaryElectionReturns': False}
 ```

Test 3:

**Review by VirginiaDooley — CHANGES_REQUESTED** [review]
Posting test results for transparency and follow up - overall looks really good!👏 

Test 1: 
Screenshot 2023-06-02 at 11 24 31 AM


Test 2: 
```
In [6]: Party.objects.last().ec_data
Execution time: 0.004854s [Database: default]
Out[6]: 
{'Id': 344,
 'Town': 'London',
 'ECRef': 'PP344',
 'Line1': '50 Iverson Road',
 'Line2': 'West Hampstead',
 'Line3': None,
 'Line4': None,
 'County': None,
 'Country': 'United Kingdom',
 'Officers': [{'Name': 'Ms Eve Rea Juan',
   'Role': 'Leader',
   'RegulatedEntityId': 344,
   'RegulatedEntityRoleId': 1},
  {'Name': 'Ms Lumiere de Monde',
   'Role': 'Nominating Officer',
   'RegulatedEntityId': 344,
   'RegulatedEntityRoleId': 2},
  {'Name': 'Ms Lumiere de Monde',
   'Role': 'Treasurer',
   'RegulatedEntityId': 344,
   'RegulatedEntityRoleId': 3}],
 'Postcode': 'NW6 2HE',
 'LapsedDate': None,
 'RegisterId': 1,
 'IsLowerTier': None,
 'ApprovedDate': '/Date(1061852400000)/',
 'PartyEmblems': [],
 'ReferendumId': None,
 'RegisterName': 'Great Britain',
 'ReferendumName': None,
 'AccountingUnits': [],
 'CampaigningName': None,
 'EntityStatusName': None,
 'FinancialYearEnd': '31/12',
 'IsGibraltarParty': False,
 'PartyDescriptions': [],
 'ReferendumOutcome': None,
 'RegulatedEntityId': 344,
 'ThirdPermStatusId': None,
 'DesignationStatusId': None,
 'RegulatedEntityName': 'www.xat.org [De-registered 01/11/07]',
 'ThirdPermStatusName': None,
 'AssetLiabilityStatus': 0,
 'RegistrationStatusId': 2,
 'DesignationStatusName': None,
 'RegulatedEntityTypeId': 1,
 'ES_AnalysedQueryString': 'PP344 www.xat.org [De-registered 01/11/07] PP  Ms Eve Rea Juan Ms Lumiere de Monde Ms Lumiere de Monde',
 'RegisterNameMinorParty': None,
 'RegistrationStatusName': 'Deregistered',
 'RegulatedEntityStatusId': 16,
 'RegulatedEntityTypeName': 'Political Party',
 'ThirdPartyParticipators': None,
 'AssetLiabilityStatusName': 'N/A',
 'CompanyRegistrationNumber': None,
 'ES_NotAnalyzedQueryString': '"pp344" "www.xat.org_[de-registered_01/11/07]" "pp"  "ms_eve_rea_juan" "ms_lumiere_de_monde" "ms_lumiere_de_monde"',
 'FieldingCandidatesInWales': True,
 'LastIsLowerTierChangeDate': None,
 'RegulatedEntityStatusName': 'Statutorily Deregistered',
 'FieldingCandidatesInEurope': False,
 'FieldingCandidatesInEngland': True,
 'RegulatedEntityAbbreviation': 'PP',
 'FieldingCandidatesInScotland': True,
 'FieldingCandidatesMinorParty': False,
 'RegulatedEntityAlternateName': None,
 'RegulatedEntityTypeShortcode': 'PP',
 'TargetedSpendingThirdParties': [],
 'DisplayNIPartyReturnExemptions': False,
 'DateLastPartyNameChangeApproved': '/Date(1193875200000)/',
 'ExemptFromQuarterlyDonationReturns': True,
 'ExemptFromQuarterlyTransactionReturns': False,
 'ExemptFromParliamentaryElectionReturns': False}
 ```

Tests 3 & 4: 
I wasn't sure how to do manually add an emblem with sql so I saved the last emblem in the db to the first party in the db, which included a description. I'm sure it's ok, but I would be feel better following the same steps you did to manually add. Can you update the testing instructions above? 🙏

**Comment by Bekabyx** [comment]
> Tests 3 & 4: I wasn't sure how to do manually add an emblem with sql so I saved the last emblem in the db to the first party in the db, which included a description. I'm sure it's ok, but I would be feel better following the same steps you did to manually add. Can you update the testing instructions above? 🙏

Sure thing! I've updated those now. Of course, you may need to tweak it depending on which party you're testing with, but this is the exact SQL I ran on my local DB. Let me know if you have any questions or want any more information and I'll do my best to update 😁

**Review by VirginiaDooley — APPROVED** [review]

File: ynr/apps/parties/tests/test_models.py [hunk 1]
```diff
@@ -24,17 +24,16 @@ def test_party_emblem(self):
         party = PartyFactory()
         PartyEmblemFactory.create_batch(3, party=party)
         self.assertEqual(party.emblems.count(), 3)
-        print(party.emblems.first().image.url)
-        self.assertTrue(
-            party.emblems.first().image.url.startswith(
-                "{}/emblems/PP0/0_example".format(settings.MEDIA_ROOT)
-            )
+        first_emblem = party.emblems.first()
+        print(first_emblem.image.url)
```
Comment (symroe) [inline ynr/apps/parties/tests/test_models.py:28]: Can you remove this print?

**Review by symroe — APPROVED** [review]
One minor comment, otherwise this is looking grand :+1:

File: ynr/apps/parties/tests/test_models.py (hunk 1 above)
Comment (Bekabyx) [reply ynr/apps/parties/tests/test_models.py]: Great spot, removed in my most recent push :)

**Pull request merged**